database/service_instance.go

```diff
@@ -0,0 +1,563 @@
+	// The status of the service instance
+	Status ServiceInstanceState `json:"status"`
+	// The billing frequency of the service instance; either MONTHLY or HOURLY.
+	SubscriptionType ServiceInstanceSubscriptionType `json:"subscriptionType"`
+	// The time zone of the operating system.
+	Timezone string `json:"timezone"`
+	// For service instances hosting an Oracle RAC database, the size in GB of the storage shared
+	// and accessed by the nodes of the RAC database.
+	TotalSharedStorage int `json:"total_shared_storage"`
+	// The Oracle Database version on the service instance.
+	Version string `json:"version"`
+}
+
+type CreateServiceInstanceInput struct {
+	// Free-form text that provides additional information about the service instance.
+	// Optional.
+	Description string `json:"description,omitempty"`
+	// Database edition for the service instance:
+	// If you specify SE, a Standard Edition 2 database is created if you specify 12.2.0.1 or 12.1.0.2
+	// for version and a Standard Edition One database is created if you specify 11.2.0.4 for version.
+	// Edition must be Enterprise Edition - Extreme Performance to configure the Database
+	// Cloud Service instance as Cluster Database.
+	// Required.
+	Edition ServiceInstanceEdition `json:"edition"`
+	// Service level for the service instance
+	// Required.
+	Level ServiceInstanceLevel `json:"level"`
+	// Array of one JSON object that specifies configuration details of the services instance.
+	// This array is not required if the level value is BASIC.
+	// Required if level value is PAAS.
+	Parameters []Parameter `json:"parameters,omitempty"`
+	// Name of Database Cloud Service instance. The service name:
+	// Must not exceed 50 characters.
+	// Must start with a letter.
+	// Must contain only letters, numbers, or hyphens.
+	// Must not contain any other special characters.
+	// Must be unique within the identity domain.
+	// Required.
+	Name string `json:"serviceName"`
+	// Desired compute shape. A shape defines the number of Oracle Compute Units (OCPUs) and amount
+	// of memory (RAM).
+	// Required.
+	Shape ServiceInstanceShape `json:"shape"`
+	// Billing unit. Valid values are:
+	// HOURLY: Pay only for the number of hours used during your billing period. This is the default.
+	// MONTHLY: Pay one price for the full month irrespective of the number of hours used.
+	// Required.
+	SubscriptionType ServiceInstanceSubscriptionType `json:"subscriptionType"`
+	// Oracle Database software version
+	// Required.
+	Version ServiceInstanceVersion `json:"version"`
+	// Public key for the secure shell (SSH). This key will be used for authentication when
+	// connecting to the Database Cloud Service instance using an SSH client. You generate an
+	// SSH public-private key pair using a standard SSH key generation tool. See Connecting to
+	// a Compute Node Through Secure Shell (SSH) in Using Oracle Database Cloud Service.
+	// Required.
+	VMPublicKey string `json:"vmPublicKeyText"`
```

Comment:
Would it be worthwhile to just omit timeout for the ServiceInstance struct and just use it in the client?